pkg/state/state.go

```diff
@@ -3248,93 +3248,136 @@ func (c MissingFileHandlerConfig) resolveFileOptions() []resolveFileOption {
+				return nil, fmt.Errorf("glob patterns in release values and secrets is not supported yet. please submit a feature request if necessary")
 			}
 			path := paths[0]
 
 			yamlBytes, err := st.RenderReleaseValuesFileToBytes(release, path)
 			if err != nil {
-				return generatedFiles, fmt.Errorf("failed to render values files \"%s\": %v", typedValue, err)
+				return nil, fmt.Errorf("failed to render values files \"%s\": %v", typedValue, err)
 			}
 
-			valfile, err := createTempValuesFile(release, yamlBytes)
-			if err != nil {
-				return generatedFiles, err
+			if err := yaml.Unmarshal(yamlBytes, &fileValues); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal values file \"%s\": %v", typedValue, err)
 			}
-			defer func() {
-				_ = valfile.Close()
-			}()
 
-			if _, err := valfile.Write(yamlBytes); err != nil {
-				return generatedFiles, fmt.Errorf("failed to write %s: %v", valfile.Name(), err)
+		case map[any]any, map[string]any:
+			if m, ok := typedValue.(map[string]any); ok {
+				fileValues = m
+			} else {
+				fileValues = make(map[string]any)
+				for k, v := range typedValue.(map[any]any) {
+					if strKey, ok := k.(string); ok {
+						fileValues[strKey] = v
+					}
+				}
 			}
 
-			st.logger.Debugf("Successfully generated the value file at %s. produced:\n%s", path, string(yamlBytes))
+		default:
+			return nil, fmt.Errorf("unexpected type of value: value=%v, type=%T", typedValue, typedValue)
+		}
 
-			generatedFiles = append(generatedFiles, valfile.Name())
-		case map[any]any, map[string]any:
-			valfile, err := createTempValuesFile(release, typedValue)
-			if err != nil {
-				return generatedFiles, err
-			}
-			defer func() {
-				_ = valfile.Close()
-			}()
+		for k, v := range fileValues {
+			mergedRaw[k] = mergeAppendValues(mergedRaw[k], v)
+		}
+	}
 
-			encoder := yaml.NewEncoder(valfile)
-			defer func() {
-				_ = encoder.Close()
-			}()
+	processor := yaml.NewAppendProcessor()
+	processed, err := processor.ProcessMap(mergedRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to process key+ syntax: %w", err)
+	}
 
-			if err := encoder.Encode(typedValue); err != nil {
-				return generatedFiles, err
-			}
+	if len(processed) > 0 {
+		valfile, err := createTempValuesFile(release, processed)
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			_ = valfile.Close()
+		}()
 
-			generatedFiles = append(generatedFiles, valfile.Name())
-		default:
-			return generatedFiles, fmt.Errorf("unexpected type of value: value=%v, type=%T", typedValue, typedValue)
+		encoder := yaml.NewEncoder(valfile)
+		defer func() {
+			_ = encoder.Close()
+		}()
+
+		if err := encoder.Encode(processed); err != nil {
+			return nil, err
 		}
+
+		generatedFiles = append(generatedFiles, valfile.Name())
 	}
+
 	return generatedFiles, nil
 }
 
+// mergeAppendValues merges two values for the same key, preserving key+ keys for later processing
+func mergeAppendValues(existing, incoming any) any {
```

Comment:
could you add unittest for this func?

Reply:
Added a unit test for it, is it suitable?

Reply:
yeah